### DISCUSSION TITLE
Using Google SMTP with Supabase Custom SMTP

### DISCUSSION THREAD

**Discussion by kangmingtay** [supabase/supabase#19646] [Troubleshooting]
_This is a copy of a troubleshooting article on Supabase's docs site. It may be missing some details from the original. View the [original article](https://supabase.com/docs/guides/troubleshooting/using-google-smtp-with-supabase-custom-smtp-ZZzU4Y)._

Hey everyone, i've been hearing feedback about how challenging it can be to get google SMTP working properly with Supabase. I've tried setting this up with a trial google workspace account and this is what i've discovered.

1.  The sender email and SMTP username has to be your google workspace admin email.
2.  The SMTP password used has to be an app password. Google workspace doesn't make it easy to figure out how to create this. You need to **enable 2FA** on your workspace account first (not inside the admin console but in <https://myaccount.google.com/>)
3.  If you want to use `smtp-relay.gmail.com` only port 465 works. If you want to use `smtp.gmail.com`, you can use port 465 or 587.
    The following screenshots show what those steps look like:

## For `smtp-relay.gmail.com`

![image](/docs/img/troubleshooting/800448a1-0245-4a26-855f-3be8f44adf03.png)

(The "Require TLS encryption" option didn't seem to affect anything when tested)![image](/docs/img/troubleshooting/5b55ddbc-abfc-4c66-a012-b8e4d23cd438.png)

## For `smtp.gmail.com`

![image](/docs/img/troubleshooting/ade24c85-50b5-47ca-a2ce-c3f77a0e128a.png)

A lot of this was figured out through trial and error.
This is unlikely to be a Supabase Auth bug because Supabase Auth uses the [native Golang SMTP library ](https://pkg.go.dev/net/smtp). If you still have issues setting up Google SMTP, you can try switching to one of these SMTP providers:

*   [Resend](https://resend.com/blog/how-to-configure-supabase-to-send-emails-from-your-domain)
*   [SendGrid](https://sendgrid.com/en-us)
*   [Mailgun](https://www.mailgun.com/)
*   AWS SES SMTP

**Comment by dr-starchildren** [comment]
Very much appreciated these instructions. I had been using Google's SMTP service since January based on them. 

I wanted to let people know about an error that I had starting a couple days ago. My signup functionality stopped working and it was only obvious in supabase logs.
In supabase logs, I had this error:
ERROR  /signup | database error on committing or rolling back transaction: write failed: write tcp 127.0.0.1:35174->127.0.0.1:5432: write: broken pipe

After troubleshooting for several hours, I narrowed it down to a SMTP auth error. When I swapped SMTP providers everything worked fine (after many other failed attempts to solve the issue). No idea why my account for Google just stopped working, but if anyone encounters this error, hopefully it'll save you time.

**Reply by TheOtherBrian1** [reply]
Hi @dr-starchildren, I believe the [issue](https://github.com/supabase/auth/issues/1502) was resolved in this [PR](https://github.com/supabase/auth/pull/1505)

**Comment by mehdinathani** [comment]
can i use my free gmail account?

**Comment by vianneypercq** [comment]
Very useful tutorial. Thank you very much, it works perfectly for my project.

**Reply by sagar98cyber** [reply]
You are using GOOGLE SMTP with SUPABASE in Production?  
How is that working out for you.

**Comment by infinia-yzl** [comment]
Thanks to this guide, I was able to set it up successfully in 2024.
* Used smtp-relay.gmail.com.
* Set up 2FA.
* Created App Password after enabling 2FA.
* Used App Password in Supabase settings.
* I set up an alternate email alias: `no-reply@domainname.com` and configured the "Sender Email" in Supabase settings to `no-reply@domainname.com`. Not sure if the alternate email alias is necessary, but I did it anyway.

**Reply by 00fff** [reply]
Hello i am having issues setting mine up currently. i keep getting "message": "Error sending confirmation mail". Could you explain how you got Google Smtp to work for you?

**Reply by marloncommona** [reply]
<img width="606" alt="image" src="https://github.com/user-attachments/assets/dce56dc9-4653-4382-b134-bbc58f949393">
Is this working for you with the Auth services of supabase?
I replicated everything and did a successful test with this tool: https://www.gmass.co/smtp-test
But it does not work with the user invite feature from supabase.

**Reply by marloncommona** [reply]
strangely I only got it to work using smtp.gmail.com ✌️

**Reply by alengcm** [reply]
> strangely I only got it to work using smtp.gmail.com ✌️

were you able to get it to reply using noreply@domain.com? I am only getting it to send from the email address I use to login. Even though sender email is different in supabase config.

**Reply by alengcm** [reply]
> > strangely I only got it to work using smtp.gmail.com ✌️
> 
> were you able to get it to reply using [noreply@domain.com](mailto:noreply@domain.com)? I am only getting it to send from the email address I use to login. Even though sender email is different in supabase config.

I used smtp-relay.gmail.com and now it is sending from no-reply@domain.com

**Reply by pabloasc** [reply]
same solution for me, I am using smtp.gmail.com now!

**Comment by oussamachah2020** [comment]
it works, but the email template even tho i changed it, it is not applied

**Comment by rotimi-best** [comment]
Quick question: Why am I required to add a custom SMTP to supabase when I don't use Supabase email verification?

**Comment by arthur-adriansens** [comment]
Hi! It didn't work for me, but when I finally tried changing the port from 465 to **587**, it worked! **465 does not work** for non-google workspace account (free basic accounts)!

**Reply by pratham-garg-456** [reply]
This worked for me too

**Comment by alvaldi-atom** [comment]
Really appreciate this guide - thank you for putting it together!

**Comment by f1yingbanana** [comment]
For people that can't get smtp-relay.google.com working, you gotta follow [this Google guide](https://support.google.com/a/answer/2956491?hl=en&src=supportwidget0&authuser=0) and set up your SMTP relay option in your Google Admin first! :)

Once you set up a Google group (e.g. noreply), then you can add that group as an alias in Gmail, then specify the group's email in Supabase.

**Reply by mishchief** [reply]
You're the GOAT thanks so much!

@kangmingtay do you think you can add some of the additional information to the original post?

I would be more than happy to summarize it for you

**Reply by kepler** [reply]
Thanks for the pointer.

Quick caveat: it's smtp-relay.**gmail**.com, not smtp-relay.**google**.com (as per the linked instructions).
